pkg/services/user/userimpl/user.go

```diff
@@ -594,3 +597,25 @@ func (s *Service) supportBundleCollector() supportbundles.Collector {
 		Fn:                collectorFn,
 	}
 }
+
+func hashUserIdentifier(identifier string, secret string) string {
+	key := []byte(secret)
+	h := hmac.New(sha256.New, key)
+	h.Write([]byte(identifier))
+	return hex.EncodeToString(h.Sum(nil))
+}
+
+func buildUserAnalyticsSettings(signedInUser user.SignedInUser, intercomSecret string) user.AnalyticsSettings {
+	var settings user.AnalyticsSettings
+
+	if signedInUser.ExternalAuthID != "" {
+		settings.Identifier = signedInUser.ExternalAuthID
+	} else {
+		settings.Identifier = signedInUser.Email + "@" + setting.AppUrl
```

Comment:
re-thinking about this, I'm not sure we gain any value by adding the `AppUrl` to this. It's probably just safe to use Email as Identifier when we don't have the ExternalAuthID. Which could simplify this code back down.

unless you can think of a security reason to try to treat the user with the same email as different.